subsys/logging/backends/log_backend_rpmsg.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2025 Lexmark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log_backend.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/logging/log_output.h>
+#include <zephyr/logging/log_backend_std.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/sys/__assert.h>
+
+#include <openamp/rpmsg.h>
+
+/**
+ * @brief Backend data structure
+ * Contains backend-specific runtime data
+ */
+struct log_backend_rpmsg_data {
+	uint32_t log_format_current;
+	struct rpmsg_endpoint ept;
+};
+
+/**
+ * @brief Backend context structure
+ * Contains references to output handler and backend data
+ */
+struct log_backend_rpmsg_ctx {
+	const struct log_output *output;
+	struct log_backend_rpmsg_data *data;
+};
+
+/**
+ * @brief Character output function
+ * This function is called by the log output subsystem to send data
+ *
+ * @param data Pointer to data buffer
+ * @param length Number of bytes to output
+ * @param ctx Context pointer (log_backend_rpmsg_ctx)
+ * @return Number of bytes actually output
+ */
+static int char_out(uint8_t *data, size_t length, void *ctx)
+{
+	const struct log_backend_rpmsg_ctx *backend_ctx = ctx;
+	struct log_backend_rpmsg_data *backend_data = backend_ctx->data;
+	int ret = 0;
+
+	if (IS_ENABLED(CONFIG_LOG_BACKEND_RPMSG_BLOCKING_MODE)) {
```

Comment:
Please check whether `CONFIG_LOG_MODE_IMMEDIATE` can even be used when `CONFIG_LOG_BACKEND_RPMSG_BLOCKING_MODE` is enabled. In that cause, I think some build warning (or Kconfig handling) may be necessary.